src/maps.rs

```diff
@@ -119,9 +120,15 @@ fn parse_maps_line<'line>(line: &'line str, pid: Pid) -> Result<MapsEntry, Error
     Ok(entry)
 }
 
-fn parse_file<R>(reader: R, pid: Pid) -> Result<Vec<MapsEntry>, Error>
+/// Parse a proc maps file from the provided reader.
+///
+/// `filter` is a filter function (similar to those usable on iterators)
+/// that determines which entries we keep (those for which it returned
+/// `true`) and which we discard (anything `false`).
+fn parse_file<R, F>(reader: R, pid: Pid, mut filter: F) -> Result<Vec<MapsEntry>, Error>
```

Comment:
how hard would it be to turn this into a proper iterator, which I think what you ultimately want. It will also allow early break, if necessary.

Reply:
Okay, so we cannot mmap the maps files, apparently, but I moved everything over to an iterator nevertheless.